lib/ruby_smb/fscc/file_information.rb

```diff
@@ -69,25 +100,26 @@ def self.name(value)
         constants.select { |c| c.upcase == c }.find { |c| const_get(c) == value }
       end
 
-      # The FILE_NAME_INFORMATION type as defined in
-      # https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/20406fb1-605f-4629-ba9a-c67ee25f23d2
-      class FileNameInformation < BinData::Record
-        endian :little
-        uint32           :file_name_length, label: 'File Name Length',  initial_value: -> { file_name.do_num_bytes }
-        string16         :file_name,        label: 'File Name',         read_length: -> { file_name_length }
-      end
-
```

Comment:
So I originally defined this one in here. Unforunately due to name mangling, what was defined here [2.1.7 FILE_NAME_INFORMATION](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/20406fb1-605f-4629-ba9a-c67ee25f23d2) shadows that of [2.4.27 FileNameInformation](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-fscc/cb30e415-54c5-4483-a346-822ea90e1e89). Luckily the later is simply composed of the former. `2.4.27 FileNameInformation` however does have the additional piece of `CLASS_LEVEL` metadata, so I opted to move the definition to include that piece of info.